### ISSUE TITLE
Builders should have occam smoke tests

### ISSUE THREAD

**Issue by fg-j** [paketo-buildpacks/builder#43]
Currently, we test new versions of the `builder.toml` in the full builder and (soon) base builder with [smoke tests that `pack create-builder` and then `pack build` a set of sample apps](https://github.com/paketo-buildpacks/full-builder/blob/856d90c9526c7313e880230a67b60dd32cd530ec/.github/workflows/test-pull-request.yml#L34). These smoke tests are slow, not parallelized, and **do not cover all of the language families the builders support**.

Each builder should have smoke tests that test **each of the language families within the builder**. The tests should check that the expected buildpack is the one that passed detection (e.g. Ruby buildpack detects on a rackup sample app).

By implementing the tests with [occam](https://github.com/paketo-buildpacks/occam), we get the benefit that the tests can be run in parallel so we'd end up with better test coverage without slowing down our workflows.

Defining a parameterized github action that accepts a list of buildpacks/language families and`pack build`s would make it simpler to keep this testing consistent across the different builders.

**Comment by fg-j** [comment]
Closing in favor of issues on specific builder repos.
